source/ed/data.cpp

```diff
@@ -493,24 +509,34 @@ create_shape(const nt::GeographicalCoord& from,
 {
     const auto nearest_from = get_nearest(from, shape);
     const auto nearest_to = get_nearest(to, shape);
+    const auto p_from = get_nearest_projection(from, shape);
+    const auto p_to = get_nearest_projection(to, shape);
+
     if (nearest_from == nearest_to) { return {from, to}; }
 
     nt::LineString res;
     const auto range = get_smallest_range(nearest_from, nearest_to);
-    res.push_back(from);
+
+    res.push_back(p_from);
     if (range.first < range.second) {
         for (auto it = range.first; it <= range.second; ++it)
             res.push_back(*it);
     } else {
         for (auto it = range.first; it >= range.second; --it)
             res.push_back(*it);
     }
-    res.push_back(to);
+    res.push_back(p_to);
 
-    // simplification at about 3m precision
+    // simplification
     nt::LineString simplified;
     boost::geometry::simplify(res, simplified, simplify_tolerance);
 
+    // if the shape is only a point, double it in order
+    // to fake a linestring object
+    if (simplified.size() == 1) {
```

Comment:
This case should never happen, no?
Do  you have a case where it's possible?
You should have 2 points from your p_from and p_to, and simplify shouldn't simplify more than 2 points.